reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/YarnContainerManager.java

```diff
@@ -612,7 +612,9 @@ private void writeToEvaluatorLog(final String entry) throws IOException {
             new BufferedWriter(new OutputStreamWriter(fs.append(path))) :
             new BufferedWriter(new OutputStreamWriter(fs.create(path)));
     ) {
-      bw.write(entry);
+      synchronized (this) {
```

Comment:
It may be better to remove the synchronized blocks at L615 and L674 and instead make `writeToEvaluatorLog` a synchronized method.